engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/TypeOfNode.java

```diff
@@ -50,6 +54,21 @@ Object doString(String value) {
     return EnsoContext.get(this).getBuiltins().text();
   }
 
+  @Specialization
+  Object doPanicException(PanicException value) {
+    return EnsoContext.get(this).getBuiltins().panic();
+  }
+
+  @Specialization
+  Object doPanicSentinel(PanicSentinel value) {
+    return EnsoContext.get(this).getBuiltins().panic();
+  }
+
+  @Specialization
+  Object doWarning(WithWarnings value) {
+    return execute(value.getValue());
```

Comment:
@JaroslavTulach The type of the Warning is the type of the underlying value - that is OK, right? Nothing broke because of that so I'm assuming this is OK.